main.go

```diff
@@ -310,7 +313,32 @@ const (
 	sourceS3    tileSource = "S3"
 )
 
+// getAndCacheTile fetches the requested tile from S3 if it exists there, or, if
+// it doesn't exist in S3, from the backing CT log and then caches it in S3.
+// Under the hood, it collapses requests for the same tile into one single
+// request. It should be preferred over getAndCacheTileUncollapsed.
 func (tch *tileCachingHandler) getAndCacheTile(ctx context.Context, tile tile) (*entries, tileSource, error) {
+	dedupKey := fmt.Sprintf("logURL-%s-tile-%d-%d", tile.logURL, tile.start, tile.end)
+
+	type entriesAndSource struct {
+		entries *entries
+		source  tileSource
+	}
+
+	innerContents, err, _ := singleflightDo(tch.cacheGroup, dedupKey, func() (entriesAndSource, error) {
```

Comment:
It'd be pretty cool as a follow-on to collect arg3 here (`shared bool`) for metrics.

Reply:
https://github.com/letsencrypt/ctile/issues/10

Reply:
Yep, could be handy!